File: src/main/java/ph/edu/cksc/college/advweb/blog/service/PostService.java
package ph.edu.cksc.college.advweb.blog.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import ph.edu.cksc.college.advweb.blog.model.Post;

import java.util.List;
import java.util.Optional;

public interface PostService {
    Post createPost(Post post);

    Post updatePost(Post post);

    Page<Post> getPosts(String query, Pageable pageable);

    Post getPostById(long postId);

    void deletePost(long id);

    List<Post> findByUserId(long userId);

    Optional<Post> findById(long id);

}

File: react-client/src/services/PostService.js
import httpClient from "./HttpCommon";

export const listPosts = (params) => {
    return httpClient.get('/posts', { params })
};

export const createPost = (post) => {
    return httpClient.post('/posts', post)
}

export const getPostById = (postId) => {
    return httpClient.get('/posts/' + postId);
}

export const updatePost = (postId, post) => {
    return httpClient.put('/posts/' + postId, post);
}

export const deletePost = (postId) => {
    return httpClient.delete('/posts/' + postId);
}